src/PhpDoc/ResolvedPhpDocBlock.php

```diff
@@ -216,6 +220,7 @@ public function merge(array $parents, array $parentPhpDocBlocks): self
 		$result->typeAliasTags = $this->getTypeAliasTags();
 		$result->typeAliasImportTags = $this->getTypeAliasImportTags();
 		$result->assertTags = self::mergeAssertTags($this->getAssertTags(), $parents, $parentPhpDocBlocks);
+		$result->selfOutTypeTag = $this->getThisOutTag();
```

Comment:
If the method overrides a parent method, it should use the `this-out` tag from the parent? That's what all the other tags do here.

Also - selfOutTypeTag vs. getThisOutTag, consistency please :) I like `self` more.